### PR TITLE
Kafka drain timeout

### PR DESCRIPTION
**Pull request by engechas**
### Description
With the addition of an external buffer, draining the remaining data on shutdown may be a long process. This PR adds a configuration setting to the KafkaBufferConfig that allows the drain period to be configured. As part of this, a new method is added to the Buffer interface to get the drain period for a given buffer.
 
### Check List
- [x] New functionality includes testing.
- [ ] New functionality has a documentation issue. Please link to it in this PR.
  - [ ] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,5 @@ gradle/tools
 # Ignore config file generated by test
 data-prepper-main/src/test/resources/logstash-filter.yaml
 data-prepper-main/src/test/resources/logstash-conf/logstash-filter.yaml
+data-prepper-plugins/sns-sink/C:\\Work\\dlq.txt
```
Comment (dlvenable) [inline .gitignore:16]: Can we just remove these files from our build? Or remove the `sns-sink` for now?

File: data-prepper-api/src/main/java/org/opensearch/dataprepper/model/buffer/Buffer.java [hunk 2]
```diff
@@ -53,4 +54,6 @@ public interface Buffer<T extends Record<?>> {
     void checkpoint(CheckpointState checkpointState);
 
     boolean isEmpty();
+
+    Duration getDrainTimeout();
```
Comment (dlvenable) [inline data-prepper-api/src/main/java/org/opensearch/dataprepper/model/buffer/Buffer.java:58]: Maybe make this a `default` method.

```
default Duration getDrainTimeout() {
  return Duration.ZERO;
}
```

File: data-prepper-api/src/test/java/org/opensearch/dataprepper/model/buffer/AbstractBufferTest.java [hunk 3]
```diff
@@ -259,6 +260,13 @@ public void testWriteAllRuntimeException() {
         Assert.assertThrows(NullPointerException.class, () -> abstractBuffer.writeAll(testRecords, 1000));
     }
 
+    @Test
+    public void testGetDrainTimeout() {
```
Comment (dlvenable) [inline data-prepper-api/src/test/java/org/opensearch/dataprepper/model/buffer/AbstractBufferTest.java:264]: If you make this a default method, move to the interface test.

**Review by dlvenable — CHANGES_REQUESTED** [review]

File: data-prepper-api/src/main/java/org/opensearch/dataprepper/model/buffer/Buffer.java (hunk 2 above)
Comment (engechas) [reply data-prepper-api/src/main/java/org/opensearch/dataprepper/model/buffer/Buffer.java:58]: Good call, will do

File: .gitignore (hunk 1 above)
Comment (engechas) [reply .gitignore:16]: I'll see if I can track the source of these down and disable the tests. If not, I'll just remove the plugin for now

Comment (dlvenable) [reply .gitignore:16]: We don't need the plugin and it isn't working anyway. So I'm fine with just removing it. See, for example, #3448 where I removed the geoip processor.

**Review by dlvenable — APPROVED** [review]

**Review by dlvenable — APPROVED** [review]

**Review by hshardeesi — APPROVED** [review]

**Pull request merged**